README.md

```diff
@@ -16,3 +16,86 @@ The project and Python dependencies can be installed from source with:
 ```
 pip3 install .
 ```
+
+## Configuration System
+
+This project provides a configuration system that supplies options (e.g., username, password) to a command-line interface (CLI) tool.
+The configuration system follows a tiered order, allowing options to be specified and overridden from both files and command-line options.
+
+### Configuration Sources
+
+In addition to CLI options, the configuration system loads options from [JSON](https://en.wikipedia.org/wiki/JSON) files located across multiple directories.
+By default, configuration files are named `edq-config.json`.
+This value is customizable, but this document will assume the default is used.
+
+For example, a configuration file containing the `user` and `pass` options might look like this:
+```json
+{
+    "user": "alice",
+    "pass": "password123"
+}
+```
+
+The table below summarizes the configuration sources in the order they are evaluated.
+The value from the later source in the table overrides the earlier one if there are multiple sources.
+
+| Source   | Description |
+| :-----   | :---------- |
+| Global   | Configuration that is loaded from a file in a user-specific location, which is platform-dependent. |
+| Local    | Configuration that is loaded from a file in the current or nearest ancestor directory. |
+| CLI File | Configuration that is loaded from one or more explicitly provided configuration files through the CLI. |
+| CLI      | Configuration that is loaded from the command line. |
```

Comment:
I think you can remove the "Configuration that is" prefix,
and just go with "Loaded from ...".